Shopify/cli
# Added print action uniqueness validation

### WHY are these changes introduced?

Closes https://github.com/Shopify/app-ui/issues/1145

The print action [should be limited to on extension per target in an app](https://docs.google.com/document/d/1oAuys3aBv2PJTGHDEk-m_6Zky5tyJltKCzFs08tUVzA/edit#bookmark=id.qpql1s1touv2). The correct way to validate the number of extensions in an app is in core, but we are waiting on the availability of the mechanism to run app-level validations. It's on the App Foundations roadmap, but won't be ready for Editions. 

As a temporary solution, we are running validation in the CLI during `dev` and `deploy` actions, that will check for duplicate print extension targets. It will abort the action and display an error in the terminal.
 
![Screenshot 2024-05-22 at 11 25 04 AM](https://github.com/Shopify/cli/assets/7654369/295e7398-970a-4447-a5cb-9e76cb1b03da)

In addition to this, we will be adding gating to web, as noted in the linked decision log above.

### WHAT is this pull request doing?

- Adds a `validatePrintActionExtensionsUniqueness` method which runs in the `loadExtensions` method.

### How to test your changes?

- 🏠 Checkout this branch in your local CLI repo
- [Create an app with 2 print extensions](https://github.com/Shopify/app-ui/wiki/Print-extensions) in the same target
- In the CLI repo, run `cli-spin pnpm run shopify app dev --path <PATH-TO-YOUR-APP>`
  - You should see this error:
 
  ![Screenshot 2024-05-22 at 11 25 04 AM](https://github.com/Shopify/cli/assets/7654369/f8ba6bee-2170-471e-9102-fd9c29b6648f)

- In the CLI repo, run `cli-spin pnpm run shopify app deploy --path <PATH-TO-YOUR-APP>`
  - You should see the error again
- Delete one of the print extensions and confirm that dev and deploy work normally.   

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
